### PR TITLE
`% cargo clippy --fix`

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

**Review by adrien-zinger — APPROVED** [review]
Look really nice

**Review by AureliaDolo — APPROVED** [review]
On day we will be clippy approved

**Comment by AureliaDolo** [comment]
bors merge

File: massa-network/src/peer_info_database.rs [hunk 1]
```diff
@@ -605,7 +605,7 @@ impl PeerInfoDatabase {
     /// # Argument
     /// * ip : ip address of the considered peer.
     pub fn out_connection_closed(&mut self, ip: &IpAddr) -> Result<(), NetworkError> {
-        let peer = self.peers.get_mut(ip).ok_or_else(|| {
+        let peer = self.peers.get_mut(ip).ok_or({
```
Comment (damip) [inline massa-network/src/peer_info_database.rs:608]: sometimes it turns ok_or_else into ok_or, sometimes it does the opposite

File: massa-pool/src/operation_pool.rs [hunk 2]
```diff
@@ -325,9 +325,11 @@ impl OperationPool {
                 .map(|op_id| {
                     self.ops
                         .get(op_id)
-                        .ok_or(PoolError::ContainerInconsistency(
-                            "op in ops by address is not in ops".to_string(),
-                        ))
+                        .ok_or_else(|| {
```
Comment (damip) [inline massa-pool/src/operation_pool.rs:328]: sometimes it turns ok_or_else into ok_or, sometimes it does the opposite

File: massa-network/src/peer_info_database.rs (hunk 1 above)
Comment (damip) [reply massa-network/src/peer_info_database.rs:608]: Any idea why ?

Comment (AureliaDolo) [reply massa-network/src/peer_info_database.rs:608]: > Arguments passed to ok_or are eagerly evaluated; if you are passing the result of a function call, it is recommended to use ok_or_else, which is lazily evaluated.

https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or

File: massa-pool/src/operation_pool.rs (hunk 2 above)
Comment (yvan-sraka) [reply massa-pool/src/operation_pool.rs:328]: https://rust-lang.github.io/rust-clippy/master/#unnecessary_lazy_evaluations

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Pull request merged**